cmake/partition_manager.cmake

```diff
@@ -117,16 +117,27 @@ else()
     )
 endif()
 
-# Add the dynamic partition as an image partition.
-set_property(GLOBAL PROPERTY
+# Check if the dynamic partition image hex has already been defined
```

Comment:
We should add a note to https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/scripts/partition_manager/partition_manager.html#hex-files explaining this special handling.

Reply:
Isn't that already documented. It says there that you can specify app_PM_HEX_FILE to point to a hex file for the partition. But since the "app" has its hex file specified here then the order of the assignment matter.
This commit removes the dependency of the order by allowing the assignment to be done before this script is executed.